### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by diabonas**
A couple of small things I came across. The most noteworthy is d6a16d361d9d7cc229c8a65da5be0d595dadd5f8, which replaces `AC_GNU_SOURCE` by `AC_USE_SYSTEM_EXTENSIONS` since the former [is deprecated](https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/Obsolete-Macros.html). Furthermore, the macro is moved before `AC_PROG_CC` as it currently creates many warnings of the form
```
configure.ac:41: warning: AC_COMPILE_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
../../lib/autoconf/specific.m4:314: AC_GNU_SOURCE is expanded from...
```
during `./bootstrap`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -101,7 +101,7 @@ script:
   - make -j$(nproc) check
   - cat test-suite.log
   - make -j$(nproc) distcheck
-  - cat config.log $(find -name test-suit.log) || true
+  - cat test-suite.log config.log
```
Comment (AndreasFuchsSIT) [inline .travis.yml]: Hmmm... this used to fail with a test-suit.log not found on travis...
Also I wanted to get the test-suit.log from the dist-check directory, not the current directory...

File: README.md [hunk 2]
```diff
@@ -41,8 +41,8 @@ it will be available again on the next boot.
 # Build and install instructions
 Standard installation using
 ```
-./bootstrap
-./configure
+bootstrap
```
Comment (AndreasFuchsSIT) [inline README.md]: For those envs where `.` is not in $PATH these commands will fail.
I'd like to keep the `./` for bootstrap and configure.
The rest below is ok...

File: .travis.yml (hunk 1 above)
Comment (diabonas) [reply .travis.yml]: Ah, got it - if `make distcheck` is successful, the `test-suite.log` is cleaned by `make distclean`, so there is nothing to show at all (well, the log from `make check` is still there, but we don't need to display it twice). If the distcheck fails and the file is not cleaned up, it is in `tpm2-totp-$PACKAGE_VERSION/_build/sub/`, which is what I `cat` in the updated commit.

File: README.md (hunk 2 above)
Comment (diabonas) [reply README.md]: Indeed, fixed.

**Pull request merged**